submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,329 @@
+        <label for="male" class="radio-lable">Male</label>
+        <input type="radio" id="female" name="sort-gender">
+        <label for="female" class="radio-lable">female</label>
+        <input type="radio" id="allGender" name="sort-gender">
+        <label for="allGender" class="radio-lable">all</label>
+        </div>
+      </div>
+
+      <div class="filter filter-name">
+        <p>sort by name</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="nameAZ" name="sort-name">
+        <label for="nameAZ" class="radio-lable">A - Z</label>
+        <input type="radio" id="nameZA" name="sort-name">
+        <label for="nameZA" class="radio-lable">Z - A</label>
+        <input type="radio" id="nameAll" name="sort-name">
+        <label for="nameAll" class="radio-lable">default</label>
+        </div>
+      </div>
+      
+      <div class="filter filter-age">
+        <p>sort by age</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="age09" name="sort-age">
+        <label for="age09" class="radio-lable">young - old</label>
+        <input type="radio" id="age90" name="sort-age">
+        <label for="age90" class="radio-lable">old - young</label>
+        <input type="radio" id="ageAll" name="sort-age">
+        <label for="ageAll" class="radio-lable">default</label>
+        </div>
+      </div>
+
+      
+      <div class="filter filter-search">
+        <p>Search user by email</p>
+        <input id="nameInput" type="text" size="30" class="searchEmail icon search">
+      </div>
+    `;
+  }
+
+  getFilters() {
+    this.filtersContainer.insertAdjacentHTML('beforeend', this.getHtmlFilter());
+  }
+}
+
+class Filter {
+  constructor() {
+    this.profilesContainer = document.querySelector('.profile-cards');
+    new Render().getFilters();
+    this.filtersPanel = document.querySelector('#filters');
+    this.searchInput = document.querySelector('#nameInput');
+    this.startFilter();
+    this.sortByNameTypes = {
+      type1: 'nameAZ',
+      type2: 'nameZA',
+      type3: 'nameAll',
+    };
+    this.sortByAgeTypes = {
+      type1: 'age09',
+      type2: 'age90',
+      type3: 'ageAll',
+    };
+  }
+
+  unckeckedRadioButton(nodeName) {
+    const activeRadioButtons = document.querySelectorAll(
+      'input[type=radio]:checked'
+    );
+    const checkedRadio = Array.from(activeRadioButtons).find(
+      node => node.name === nodeName
+    );
+    if (checkedRadio) checkedRadio.checked = false;
+  }
+
+  selectSort(selected) {
+    const profileNames = document.querySelectorAll('.user-mail');
+    this.searchInput.addEventListener('input', ({ target }) => {
+      let filter = target.value.toUpperCase();
+      profileNames.forEach(nameNode => {
+        if (nameNode.textContent.toLocaleUpperCase().startsWith(filter)) {
+          nameNode.closest('.user-card').hidden = false;
+        } else {
+          nameNode.closest('.user-card').hidden = true;
+        }
```

Comment:
You working on DOM itself which is crunchy. Better try to re-render array of filtered users.

Reply:
Got it! According to your recommendation I save data to array and then work with it, sorted and then render